### DISCUSSION TITLE
MQTT naming changes announce for Homeassistant

### DISCUSSION THREAD

**Discussion by mesistostefre** [QuantumEntangledAndy/neolink#135] [General]
Here:
 [https://www.home-assistant.io/blog/2023/08/02/release-20238/#breaking-changes](https://www.home-assistant.io/blog/2023/08/02/release-20238/#breaking-changes)
a change to the MQTT entity naming has been announced.

In fact, version 2023.8 throws a waring for neolink entities created. Like for sensor.camera_battery.

I guess this needs a code change ... unless there is a way to configure differently in the [cameras.mqtt] section on neolink.

Any idea?

**Comment by QuantumEntangledAndy** [comment]
If it's deprecated we will just have to update it. I don't use HA much, do you think you can tackle the changes for this?

**Reply by QuantumEntangledAndy** [reply]
I think this is the relevant part of the change log:

---

The way MQTT entities are named and how device configuration can be shared between discovered entities has changed.

### Sharing of device configuration

Discovered MQTT entities can share device configuration, meaning one entity can include the full device configuration, and other entities can link to that device by only setting mandatory fields.

The mandatory fields were previously limited to at least one of connection and identifiers, but have now been extended to at least one of connection and identifiers as well as the name.

### Naming of MQTT entities

The naming of MQTT entities has been changed to be aligned with the entity naming guidelines:

The has_entity_name entity will be set to True on all MQTT entities.
This means any MQTT entity which is part of a device will automatically have its friendly_name attribute prefixed with the device name.
Unnamed binary_sensor, button, event, number, and sensor entities will now be named by their device class instead of being named “MQTT binary sensor” etc.
It’s now allowed to set an MQTT entity’s name to None to mark it as the main feature of a device.

**Reply by mesistostefre** [reply]
Yep, the "Naming of ..." section.

I'll try to figure it out one back from vacation... ;-)

**Comment by mesistostefre** [comment]
The good news is: The error message reads much worse than it actually seems to be. Worst case the entity will be created twice in HA. To be expected: "friendly name" in HA will look a bit awkward.
